server/webhook.go

```diff
@@ -514,12 +514,33 @@ func (s *Server) readFileContent(ctx context.Context, pushEvent *vcs.PushEvent,
 
 // prepareIssueFromPushEventSDL returns the migration info and a list of update
 // schema details derived from the given push event for SDL.
-func (s *Server) prepareIssueFromPushEventSDL(ctx context.Context, repo *api.Repository, pushEvent *vcs.PushEvent, schemaInfo map[string]string, file, webhookEndpointID string) (*db.MigrationInfo, []*api.UpdateSchemaDetail) {
+func (s *Server) prepareIssueFromPushEventSDL(ctx context.Context, repo *api.Repository, pushEvent *vcs.PushEvent, schemaInfo map[string]string, file string, fileType fileItemType, webhookEndpointID string) (*db.MigrationInfo, []*api.UpdateSchemaDetail) {
 	if schemaInfo == nil {
-		log.Debug("Ignored non-schema file for SDL",
-			zap.String("file", file),
-		)
-		return nil, nil
+		// NOTE: We do not want to use filepath.Join here because we always need "/" as the path separator.
+		migrationInfo, err := db.ParseMigrationInfo(file, path.Join(repo.BaseDirectory, repo.FilePathTemplate))
+		if err != nil {
+			log.Error("Failed to parse migration info",
+				zap.Int("project", repo.ProjectID),
+				zap.Any("pushEvent", pushEvent),
+				zap.String("file", file),
+				zap.Error(err),
+			)
+			return nil, nil
+		}
+
+		// We only allow DML files when the project uses the state-based migration
+		if migrationInfo.Type != db.Data {
```

Comment:
The data change part should be consistent between SDL and DDL projects.

Reply:
Hmm, why part is not consistent here?

Reply:
Added a code comment in https://github.com/bytebase/bytebase/pull/2628/commits/db56badff3e95aaaa945586335f8387a5567e452 per discussion in Lark, PTAL!